test/config.yaml

```diff
@@ -38,123 +36,107 @@ receivers:
-        - action: add_label
-          new_label: app_id
-          new_value: ${env:HEROKU_APP_ID}
-        - action: add_label
-          new_label: app_name
-          new_value: ${env:HEROKU_APP_NAME}
-  # Optional: The following processor can be used to add a default "deployment.environment" attribute to the logs and 
-  # traces when it's not populated by instrumentation libraries.
-  # If enabled, make sure to enable this processor in the pipeline below.
-  #resource/add_environment:
-    #attributes:
-      #- action: insert
-        #key: deployment.environment
-        #value: staging/production/...
+      - include: .*
+        match_type: regexp
+        action: update
+        operations:
+          - action: add_label
+            new_label: dyno_id
+            new_value: ${env:HEROKU_DYNO_ID}
+          - action: add_label
+            new_label: app_id
+            new_value: ${env:HEROKU_APP_ID}
+          - action: add_label
+            new_label: app_name
+            new_value: ${env:HEROKU_APP_NAME}
+          # Optional: The following processor can be used to add a default "deployment.environment" attribute to the logs and
+          # traces when it's not populated by instrumentation libraries.
+          # If enabled, make sure to enable this processor in the pipeline below.
+          #resource/add_environment:
+          #attributes:
+          #- action: insert
+          #key: deployment.environment
+          #value: staging/production/...
 
 exporters:
   # Traces
   sapm:
     access_token: "${SPLUNK_ACCESS_TOKEN}"
-    endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace"
+    endpoint: "${SPLUNK_TRACE_URL}"
   # Metrics + Events
   signalfx:
     access_token: "${SPLUNK_ACCESS_TOKEN}"
-    realm: "${SPLUNK_REALM}"
-    # Use instead when sending to gateway
-    #api_url: http://${SPLUNK_GATEWAY_URL}:6060
-    #ingest_url: http://${SPLUNK_GATEWAY_URL}:9943
+    api_url: "${SPLUNK_API_URL}"
+    ingest_url: "${SPLUNK_INGEST_URL}"
+  splunk_hec/profiling:
+    token: "${SPLUNK_ACCESS_TOKEN}"
+    endpoint: "${SPLUNK_PROFILING_URL}"
   # Send to gateway
   #otlp:
   # endpoint: "${SPLUNK_GATEWAY_URL}:4317"
   # insecure: true
-  # Debug
+
+  # Debug. Add this to the pipeline you want to debug.
   logging:
     loglevel: debug
 
 service:
-  extensions: [health_check, http_forwarder, zpages]
+  extensions: [health_check, http_forwarder, zpages, memory_ballast]
   pipelines:
     traces:
-      receivers: [jaeger, otlp, smartagent/signalfx-forwarder, zipkin]
+      receivers: [jaeger, otlp, zipkin]
       processors:
-      - memory_limiter
-      - batch
-      - attributes/heroku
+        - memory_limiter
+        - batch
+        - attributes/heroku
       #- resource/add_environment
       exporters: [sapm, signalfx]
-      # Use instead when sending to gateway
-      #exporters: [otlp, signalfx]
     metrics:
-      receivers: [otlp, signalfx, smartagent/heroku-metadata, smartagent/signalfx-forwarder]
```

Comment:
Why do we remove `smartagent/heroku-metadata` and `smartagent/signalfx-forwarder`? I'm not sure what `smartagent/heroku-metadata` does but `smartagent/signalfx-forwarder` should be needed as long as our instrumentations still use that

Reply:
We remove them to allow the user to not install the smart agent bundle. See https://github.com/signalfx/splunk-otel-collector-heroku/issues/4 for context.

Reply:
Ok sounds good 👍

Reply:
You raise a very good point though. I'll look if we can move the signalfx forwarder out of the agent bundle into its own component so we can work without importing the bundle.

Reply:
AFAIR `signalfx forwarder` doesn't do anything other than existing trace and metrics receivers can do, it just provides one endpoint for all the formats. Ideally, instrumentation libraries should stop using it, but I don't know what's the status, some of them might still use it